### PR TITLE
Feat/badge and pin

### PR DESCRIPTION
**Pull request by u12206050**

### CODE REVIEW

File: src/composables/variants.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export const variants = {
```
Comment (laurensgroeneveld) [inline src/composables/variants.ts:1]: This feels a bit too generally named as there might be components that only need a certain background color for example, but we can tackle that when we run into it.

**Review by laurensgroeneveld — CHANGES_REQUESTED** [review]
Looks great visually, just have some comments on code details.

File: src/components/BccBadge/BccBadge.stories.ts [hunk 2]
```diff
@@ -30,15 +33,12 @@ export default {
...
+  components: { BccBadge },
   setup() {
     return { args };
   },
   template: `
-    <BccBadge v-bind="args">
-      <template #icon>
-        <RadioButtonUncheckedIcon />
-      </template>
+    <BccBadge v-bind="args" icon="CheckCircleIcon">
```
[the remark this answers is not in the document]
Comment (u12206050) [reply src/components/BccBadge/BccBadge.stories.ts:41]: Hmm, no the icon is globally registered in the story so should only need to pass the icon name as a string. Works for me...

[the remark this answers is not in the document]
Comment (laurensgroeneveld) [reply src/components/BccBadge/BccBadge.stories.ts:41]: I say "for me", this is when checking the deployed version at https://black-dune-0dafdd603-36.westeurope.2.azurestaticapps.net/?path=/docs/components-bccbadge--example, I can check it locally.

[the remark this answers is not in the document]
Comment (laurensgroeneveld) [reply src/components/BccBadge/BccBadge.stories.ts:41]: Wow really weird, I just opened that page and then it shows, but with a reload it vanishes.

File: src/components/BccPin/BccPin.stories.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  component: BccPin,
+  argTypes: {
+    variant: {
+      description: "Determines the styling of the badge",
+      options: ["neutral", "error", "warning", "success", "info", "systemInfo"],
+      control: { type: "radio" },
+    },
+    size: {
+      description: "Size of the badge",
+      options: ["sm", "md", "lg"],
```
[the remark this answers is not in the document]
Comment (u12206050) [reply src/components/BccPin/BccPin.stories.ts:20]: Yeh, wondered if we should just keep the ones that are in the design system, even though more are supported or?

File: src/components/BccPin/BccPin.stories.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    <BccPin v-bind="args" icon="CheckCircleIcon" />
+  `,
+});
+
+export const Example = Template.bind({});
+Example.parameters = {
+  viewMode: "docs",
+};
+Example.args = {
+};
```
[the remark this answers is not in the document]
Comment (u12206050) [reply src/components/BccPin/BccPin.stories.ts:41]: Ah right, didn't know

File: src/components/BccPin/BccPin.vue [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<script setup lang="ts">
+import { sizeVariants, variants } from '../../composables/variants';
+
+withDefaults(defineProps<{
+  icon?: any
```
[the remark this answers is not in the document]
Comment (u12206050) [reply src/components/BccPin/BccPin.vue:5]: jip

File: src/components/BccPin/BccPin.stories.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (laurensgroeneveld) [reply src/components/BccPin/BccPin.stories.ts:20]: That would be my preference yes 👍 Then designers and developers work with the same toolbox.

**Review by laurensgroeneveld — APPROVED** [review]
Looking good 👍

As discussed, we'll investigate the sometimes vanishing icon on the Badge later, it's probably a Storybook peculiarity.

**Pull request merged**